docs/features/interceptors.md

```diff
@@ -179,6 +180,37 @@ This allows generator authors to avoid *polluting lookup* with interceptors, hel
 
 We may also want to consider adjusting behavior of `[EditorBrowsable]` to work in the same compilation.
 
+### Struct receiver capture
+
+An interceptor whose `this` parameter takes a struct by-reference can generally be used to intercept a struct instance method call, assuming the methods are compatible per [Signature matching](#signature-matching). This includes situations where the receiver must be implicitly captured to temp before the invocation, even if such capture is not permitted when the interceptor is called directly. See also [12.8.9.3 Extension method invocations](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#12893-extension-method-invocations) in the standard.
+
+
+```cs
+using System.Runtime.CompilerServices;
+
+struct S
+{
+    public void Original() { }
+}
+
+static class Program
+{
+    public static void Interceptor()
+    {
+        new S().Original(); // L1: interception is valid, no errors.
+        new S().Interceptor(); // error CS1510: A ref or out value must be an assignable variable
+    }
+}
+
+static class D
+{
+    [InterceptsLocation(1, "..(refers to call at L1)")]
+    public static void Interceptor(this ref S s)
+}
+```
+
+The goal of this is to avoid a "decoder ring" experience where an interceptor author needs to copy exactly the same ref kind sometimes and use a different ref kind other times, depending on the call arguments. We generally want that if your interceptor is accessible at the call site and uses all the same parameter ref kinds and types that you will be allowed to intercept.
```

Comment:
Not following the decoder ring bit because the `ref` kind is the same here.

Reply:
I unintentionally made this point confusing. Let me try to re-state it.

If we were to disallow intercepting a call like `new S().Original()` with an interceptor like `public static void Interceptor(this ref S s)`, it means one of two things:

1. If you don't own type `S`, you have no ability to intercept calls to `Original` (unless user happens to always be calling it on a compatible lvalue). This seems like an unreasonable limitation.
2. Or, we would devise some other rule which allows intercepting a call like `new S().Original()` with an extension method. For example, by permitting it if the interceptor takes its `this` parameter by-value, since a direct call to the interceptor method would be valid in that case. The "decoder ring" would be the interceptor author's decision of which ref kind to use, based on the call receiver.